features/0641-linking-binary-objects-to-credentials/README.md

```diff
@@ -0,0 +1,219 @@
+
+When issuing and verifying a credential, messages have to be send between the holder, issuer and verifier. In order to circumvent additional complexity, such as looking at previously sent credentials for the attachment, the attachments decorator, when containing an attachment, MUST be sent at all of the following steps:
+
+**Issue Credential V1 & V2**
+
+1. Credential Proposal
+2. Credential Offer
+3. Credential Request
+4. Credential
+
+**Present Proof V1 & V2**
+
+1. Presentation
+2. Presentation Request
+3. Present Proposal
+
+### Linking
+
+When a credential is issued with an attachment in the attachments decorator, be it a base64-encoded file or a hosted file, the link has to be made between the credential and the attachment. The link MUST be made with the `attribute.value` of the credential and the `@id` tag of the attachment in the attachments decorator.
+
+### Hashlink
+
+A hashlink, as specified in [IETF: Cryptographic Hyperlinks](https://tools.ietf.org/html/draft-sporny-hashlink-06), is a formatted hash that has a prefix of `hl:` and an optional suffix of metadata. The hash in the hashlink is a [multihash](https://tools.ietf.org/html/draft-multiformats-multihash-02), which means that according to the prefix of the hash it is possible to see which hashing algorithm and encoding algorithm has been chosen. An example of a hashlink would be:
+
+`hl:zQmcWyBPyedDzHFytTX6CAjjpvqQAyhzURziwiBKDKgqx6R`
+
+This example shows the prefix of `hl:` indicating that it is a hashlink and the hash after the prefix is a multihash.
+
+The hashlink also allows for opional metadata, such as; a list of URLs where the attachment is hosted and a MIME-type. These metadata values are encoded in the CBOR data format using the specified algortihm from section 3.1.2 in the [IETF: Cryptographic Hyperlinks](https://tools.ietf.org/html/draft-sporny-hashlink-06).
+
+When a holder receives a credential with hosted attachments, the holder MUST rehost these attachments. This is done in order to prevent the phone-home problem.
```

Comment:
Hmmm...is this a MUST or a MAY with a note about why it is recommended that the holder rehost.  In some use cases, many holders may not care about the phone home problem.

I'm pondering this vs. recommending this change.  I'm just wondering whether people think this will lead to holders inadvertently not rehosting the document.

Reply:
I changed it to MAY, because I agree that for some holders it might not be relevant. I am also thinking about whether this is even an issue the framework should fix. It might just be a problem to solve for a framework consumer.